crates/sui-network/src/discovery/mod.rs

```diff
@@ -105,7 +105,19 @@ impl DiscoveryEventLoop {
                     self.handle_trusted_peer_change_event(event);
                 }
                 Some(task_result) = self.tasks.join_next() => {
-                    task_result.unwrap();
+                    match task_result {
+                        Ok(()) => {},
+                        Err(e) => {
+                            if e.is_cancelled() {
+                                // avoid crashing on ungraceful shutdown
+                            } else if e.is_panic() {
+                                // propagate panics.
+                                std::panic::resume_unwind(e.into_panic());
+                            } else {
+                                panic!("task failed: {e}");
+                            }
+                        },
+                    };
```

Comment:
I still think this is not a bug here but instead one in the benchmarking code but 🤷🏼‍♂️

Reply:
I agree this is not a bug, in some cases cancelled statuses are unexpected and maybe the code should crash. And it is better to use proper shutdown in stress and other tests. For now I don't see this change causing unexpected issues. Maybe we can come back and tighten the conditions later ...